src/main/java/com/mapr/synth/Synth.java

```diff
@@ -248,10 +269,20 @@ public static void header(Format format, List<String> names, PrintStream out) th
             }
         }
 
-        public static int generateFile(Options opts, SchemaSampler s, PrintStream out, int count) throws IOException {
-            for (int i = 0; i < count; i++) {
-                format(opts.format, opts.quote, s.getFieldNames(), s.sample(), out);
+
+        public static int generateFile(Options opts, SchemaSampler s, Template template, PrintStream out, int count) throws IOException {
+            if (template != null) {
+                PrintWriter writer = new PrintWriter(out);
```

Comment:
This is a problematic idiom here.  Better to use an InputSupplier if you want to pass in an input that you don't close.  Otherwise, use try with resources to guarantee closing of the file.

The problem is that you should never separate the code that opens from the code that closes.  The code that you pass a writer to may not adhere to your planned contract and you can't tell from a small context whether it does.